Test meshraster

pp-mo opened the pull request:
Simple tests for the basic MeshRasterize operation.

Based on a tiny cubesphere sample datafile (included).
Also fixed a bug in basic point-in-face test, to make it inclusive of the face edges.

pp-mo commented:
@lbdreyer now good to go :crossed_fingers: ?

lbdreyer commented on lib/cartopy_mesh_raster/tests/test_MeshRasterize.py:
```diff
@@ -0,0 +1,123 @@
+class TestMeshRasterize(IrisTest):
+    def setUp(self):
+        # Read in a small test file + interpret as mesh data.
+        test_filename = 'tiny_4x4_cubesphere.nc'
```
For some reason I can't do a ncdump on this file. Seems a bit concerning

lbdreyer replied:
I hadn't downloaded the file correctly. I can now ncdump the file

pp-mo replied:
Shouldn't be a problem... it works for me.
@lbdreyer is this fixed now ?

lbdreyer commented on lib/cartopy_mesh_raster/tests/test_MeshRasterize.py:
```diff
@@ -0,0 +1,123 @@
+                for var in (var_node_lats, var_node_lons))
+
+            units = var_node_lons.units
+            if 'degree' not in units:
```
Should this not be `'degrees'`?

pp-mo replied:
I was thinking that the unit string can be either singular or plural, so testing with 'in' would accept either.  You can possibly also get terms like "degrees_east" (which is still a valid unit).

lbdreyer replied:
That makes sense 👍  I thought it was just a typo :)

lbdreyer commented on lib/cartopy_mesh_raster/tests/test_MeshRasterize.py:
```diff
@@ -0,0 +1,123 @@
+                node_lats, node_lons = (np.rad2deg(arr)
+                                        for arr in (node_lats, node_lons))
+
+            face_nodes_indices = ds.variables['Mesh2_face_nodes'][:] - 1
```
Maybe worth adding a comment to say that you are doing this because it is one-based numbering

pp-mo replied:
Ok

pp-mo commented on lib/cartopy_mesh_raster/tests/test_MeshRasterize.py:
```diff
@@ -0,0 +1,123 @@
+            face_nodes_indices = ds.variables['Mesh2_face_nodes'][:] - 1
+            face_values = ds.variables['face_values'][:]
+
+        self._face_values_min_max = (np.min(face_values), np.max(face_values))
```
I think this line is obsolete -- will remove ...

pp-mo commented:
Thanks @lbdreyer 
I've added some comments to the test code, and removed an obsolete line.
I looked into using xarray to load the test data, but I couldn't see any clear benefits.
Please re-review..

lbdreyer commented:
Thanks @pp-mo !
